recipes/r-archive/0001-Force-so-extension-on-libconnection.patch

```diff
@@ -0,0 +1,28 @@
+From 26ddece8eaac32dae277dca98283b5a90f3340ea Mon Sep 17 00:00:00 2001
+From: Pat Schloss <pschloss@umich.edu>
+Date: Tue, 27 Sep 2022 17:04:32 -0400
+Subject: [PATCH] Force so extension on libconnection
+
+---
+ R/zzz.R | 6 ++++++
+ 1 file changed, 6 insertions(+)
+
+diff --git a/R/zzz.R b/R/zzz.R
+index 30f61ed..e27d9d5 100644
+--- a/R/zzz.R
++++ b/R/zzz.R
+@@ -1,5 +1,11 @@
+ .onLoad <- function(libname, pkgname) {
+   lib_path <- system.file("lib", .Platform$r_arch, paste0("libconnection", .Platform$dynlib.ext), package = "archive")
++
++  if(lib_path == ""){
++    lib_path <- system.file("lib", .Platform$r_arch, paste0("libconnection", ".so"), package = 
++"archive")
++  }
```

Comment:
Could you please provide some more context on why you created this patch? Also, below in this file the function `.onUnload()` also uses `.Platform$dynlib.ext`, so presumably it should be changed there as well

Reply:
When I am building this using the default Mac OS X environment, `.Platform$dynlib.ext` is `.so` and so it happily finds `libconnection.so` and generates a full string for `lib_path`.

```
lib_path <- system.file("lib", .Platform$r_arch, paste0("libconnection", .Platform$dynlib.ext), package = "archive")
```

However, when I build this in a conda environment, `.Platform$dynlib.ext` is `.dylib`. This leads `lib_path` to being `""`, not finding `libconnection.so` and failing to load the library.

The solution that I've come up with is to replace those lines with the co-submitted patch. I've tested this on a Mac OS X with R (>4.1) inside and outside of conda and it works. I have updated the `.onUnload()` function as well.

Reply:
Thanks for the explanation. I don't understand why `.Platform$dynlib.ext` is different inside a conda env on macOS. Hopefully someone more knowledgeable will be able to comment on the suitability of your patch. It's hard for me to anticipate any unintended consequences.